edalize/templates/vivado/vivado-makefile.j2

```diff
@@ -8,5 +8,14 @@ $(NAME).bit:  $(NAME)_run.tcl $(NAME).xpr
 $(NAME).xpr: $(NAME).tcl
 	vivado -mode batch -source $<
 
+ifdef HWTARGET
+program: $(NAME).bit $(NAME)_pgm.tcl
+	vivado -mode batch -source $(NAME)_pgm.tcl -tclargs $(HWTARGET)
+else
+program: $(NAME).bit $(NAME)_pgm.tcl
+	vivado -mode batch -source $(NAME)_pgm.tcl
+endif
```

Comment:
Why not just set `HWTARGET` as an environment variable (if it is passed as `make` variable, convert it to an environment variable here). You can then pick it up from the TCL script and don't need to rely on fragile argv parsing.